ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/AlterMaterializedViewRebuildAnalyzer.java

```diff
@@ -63,6 +66,20 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
       unparseTranslator.addTableNameTranslation(tableTree, SessionState.get().getCurrentDatabase());
       return;
     }
+
+    try {
+      Boolean outdated = db.isOutdatedMaterializedView(getTxnMgr(), tableName);
+      if (outdated != null && !outdated) {
+        String msg = String.format("Materialized view %s.%s is up to date. Cancelling rebuild.",
+                tableName.getDb(), tableName.getTable());
+        LOG.info(msg);
+        console.printInfo(msg, false);
```

Comment:
nit. `Materialized view %s.%s is up to date. Cancelling rebuild.` -> `Materialized view %s.%s is up to date. Skipping rebuild.` ?

Reply:
Changed.